service/worker/deletenamespace/reclaimresources/workflow.go

```diff
@@ -0,0 +1,170 @@
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package reclaimresources
+
+import (
+	"fmt"
+	"time"
+
+	"go.temporal.io/sdk/log"
+	"go.temporal.io/sdk/temporal"
+	"go.temporal.io/sdk/workflow"
+
+	"go.temporal.io/server/common/log/tag"
+	"go.temporal.io/server/service/worker/deletenamespace/deleteexecutions"
+	"go.temporal.io/server/service/worker/deletenamespace/errors"
+)
+
+const (
+	// WorkflowName is the workflow name.
+	WorkflowName = "temporal-sys-reclaim-namespace-resources-workflow"
+
+	checkMaxAttempts = 10
+)
+
+type (
+	ReclaimResourcesParams struct {
+		deleteexecutions.DeleteExecutionsParams
+	}
+
+	ReclaimResourcesResult struct {
+		SuccessCount int
+		ErrorCount   int
+	}
+)
+
+var (
+	retryPolicy = &temporal.RetryPolicy{
+		InitialInterval: 1 * time.Second,
+		MaximumInterval: 10 * time.Second,
+	}
+
+	localActivityOptions = workflow.LocalActivityOptions{
+		RetryPolicy:            retryPolicy,
+		StartToCloseTimeout:    10 * time.Second,
+		ScheduleToCloseTimeout: 5 * time.Minute,
+	}
+
+	deleteExecutionsWorkflowOptions = workflow.ChildWorkflowOptions{
+		RetryPolicy:              retryPolicy,
+		WorkflowRunTimeout:       60 * time.Minute,
+		WorkflowExecutionTimeout: 24 * time.Hour,
+	}
+)
+
+func validateParams(params *ReclaimResourcesParams) error {
+	if params.NamespaceID.IsEmpty() {
+		return temporal.NewNonRetryableApplicationError("namespace ID is empty", "", nil)
+	}
+
+	if params.Namespace.IsEmpty() {
+		return temporal.NewNonRetryableApplicationError("namespace is empty", "", nil)
+	}
+
+	return nil
+}
+
+func ReclaimResourcesWorkflow(ctx workflow.Context, params ReclaimResourcesParams) (ReclaimResourcesResult, error) {
+	logger := workflow.GetLogger(ctx)
+	logger.Info("Child workflow started.", tag.WorkflowType(WorkflowName))
+
+	if err := validateParams(&params); err != nil {
+		return ReclaimResourcesResult{}, err
+	}
+
+	var a *Activities
+
+	// Step 1. Delete workflow executions.
+	result, err := deleteWorkflowExecutions(ctx, params, a, logger)
+	if err != nil {
+		return result, err
+	}
+
+	// Step 2. Delete namespace.
+	ctx5 := workflow.WithLocalActivityOptions(ctx, localActivityOptions)
+	err = workflow.ExecuteLocalActivity(ctx5, a.DeleteNamespaceActivity, params.Namespace, params.NamespaceID).Get(ctx, nil)
+	if err != nil {
+		return result, fmt.Errorf("%w: DeleteNamespaceActivity: %v", errors.ErrUnableToExecuteActivity, err)
+	}
+
+	logger.Info("Child workflow finished successfully.", tag.WorkflowType(WorkflowName))
+	return result, nil
+}
+
+func deleteWorkflowExecutions(ctx workflow.Context, params ReclaimResourcesParams, a *Activities, logger log.Logger) (ReclaimResourcesResult, error) {
```

Comment:
should not take a *Activities as input.

Reply:
also no need to take logger as input. You can get logger from workflow.GetLogger(ctx)